File: 문제풀이/온라인 저지/프로그래머스/# 02. Level 2/# 012946. 하노이의 탑/kotlin/Solution.kt
class Solution {
    fun solution(n: Int): Array<IntArray> {
        val answer = ArrayList<IntArray>()
        fun hanoi(n: Int, from: Int, mid: Int, to: Int) {
            if (n == 1) {
                answer += intArrayOf(from, to)
                return
            }
            hanoi(n-1, from, to, mid)
            hanoi(1, from, mid, to)
            hanoi(n-1, mid, from, to)
        }
        hanoi(n, 1, 2, 3)
        return answer.toTypedArray()
    }

}

File: 문제풀이/온라인 저지/프로그래머스/# 02. Level 2/# 012946. 하노이의 탑/python/solution.py
def solution(n):
    answer = []

    def hanoi(n, start, mid, end):
        if n == 1:
            answer.append([start, end])
            return
        hanoi(n - 1, start, end, mid)
        hanoi(1, start, mid, end)
        hanoi(n - 1, mid, start, end)

    hanoi(n, 1, 2, 3)
    return answer

